### PR TITLE
Dtype casting mode

### PR DESCRIPTION
**Pull request by RickSanchezStoic**
Adds Dtype Casting modes.

### CODE REVIEW

File: docs/overview/deep_dive/data_types.rst [hunk 1]
```diff
@@ -1,487 +1,520 @@
-Data Types
```
Comment (VedPatwardhan) [inline docs/overview/deep_dive/data_types.rst:1]: I think the formatting issue is still not completely resolved in this file

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hey @RickSanchezStoic, please fix the formatting issue for data_types.rst as well, thanks for fixing it for the other 2 files 🙂

File: docs/overview/deep_dive/data_types.rst [hunk 2]
```diff
@@ -461,6 +461,39 @@ The same workflow has been implemented for :code:`supported_dtypes`, :code:`unsu
...
+
+`crosscast_data_types` is for cases when a function doesn't support `int` dtypes, but supports `float` and vice-versa. In such cases,
+we cast to the default supported `float` dtype if it's the unsupported integer case or we cast to the default supported `int` dtype if it's the unsupported `float` case.
+
+The `cast_data_types` mode is the combination of all the three modes that we discussed till now. It works it way from crosscasting to upcasting and finally to downcasting to provide support
+for any unsupported dtype that is encountered by the functions.
+
+Together with these modes we provide some level of flexibility to users when they encounter functions that don't support a dtype which is otherwise supported
+by the backend.
+
```
Comment (VedPatwardhan) [inline docs/overview/deep_dive/data_types.rst:495]: Can we also add a note saying that setting these modes can lead to issues with memory consumption or precision so use these modes expecting that to happen?

**Review by VedPatwardhan — APPROVED** [review]
Hey @RickSanchezStoic, one last change, feel free to merge once that's done. Thanks 🙂

File: docs/overview/deep_dive/data_types.rst (hunk 2 above)
Comment (RickSanchezStoic) [reply docs/overview/deep_dive/data_types.rst:495]: sure

**Pull request merged**